### PR TITLE
docs: Recover Sherlock documentation

### PR DESCRIPTION
**Pull request by BasixKOR**
**What's the problem this PR addresses?**
The issue template advises to use Sherlock for reproduction but the link is broken.

...

**How did you fix it?**
Recover the Sherlock documentation from `website` branch.
...

**Checklist**

- [x] I have read the [Contributing Guide](https://yarnpkg.com/advanced/contributing).

- [x] I have set the packages that need to be released for my changes to be effective.

- [x] I will check that all automated PR checks pass before the PR gets reviewed.

### CODE REVIEW

File: packages/docusaurus/docs/advanced/04-technical/sherlock.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
+---
+category: advanced
+slug: /advanced/sherlock
+title: "Sherlock"
+description: An overview of Yarn's GitHub Actions issue toolkit that helps users provide minimal reproductions.
+hidden: true
```
Comment (BasixKOR) [inline packages/docusaurus/docs/advanced/04-technical/sherlock.md:6]: I'm not sure if this does work on newer Docusarus-based website but I don't think it should be a dealbreak.

**Comment by arcanis** [comment]
I wonder if perhaps we shouldn't just remove Sherlock altogether. We don't use it much, and its playground is often broken 😞

**Comment by clemyan** [comment]
As far as I understand, the playground was essentially just an API (+ Next.js app?) hosted on CodeSandbox? What makes it difficult to maintain? What eventually completely broke it?

I think sherlock and the playground is important unless/until the contributors have enough bandwidth to triage issues/bug reports.

**Comment by BasixKOR** [comment]
Per my last exprience with Sherlock the problem was often more of 'How can move this reproduction to Sherlock?' rather than reproduction itself, so I think the issue could be potentially resolved with better doucmentation on how to use Sherlock. I assume this discussion would fit better on dedicated therad on Discussions tab?

**Comment by clemyan** [comment]
UX is definitely important for both sherlock and the playground, but I want to get the playground back up first to at least have something. I can get a PoC working on CodeSandbox but that seems like the same architecture as the old playground, so I'd like the maintainers' input on the pitfalls of maintaining the old playground before continuing.

File: packages/docusaurus/docs/advanced/04-technical/sherlock.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+```js repro
+const foo = await ...;
+```
+~~~
+
+> **Important:** Note that the codefence is not only tagged `js`, but also `repro`. This is critical, as we don't want to execute every single snippet!
+
+## Developing your reproduction
+
+The easiest and most convenient way to develop your reproduction is by using the [Playground](/playground) (powered by [CodeSandbox](https://codesandbox.io/)).
```
Comment (wojtekmaj) [inline packages/docusaurus/docs/advanced/04-technical/sherlock.md:33]: BTW: This link is also dead 💀

**Comment by BasixKOR** [comment]
Per the comment above and the PR #6306:

> I wonder if perhaps we shouldn't just remove Sherlock altogether. We don't use it much, and its playground is often broken 😞

I'll close this PR for now, please repoen when Sherlock playground and other tools become useful again!

**Pull request closed without merging**